### ISSUE TITLE
WS-Scan

### ISSUE THREAD

**Issue by stevensantos-docs** [SimulPiscator/AirSane#53]
Not an issue, but have you ever thought about doing the same thing for WS-Scan?  One server for all scanners is such a nice idea...

**Comment by SimulPiscator** [comment]
Absolutely, I agree. I actually spent some effort on adding a WS-Scan server to AirSane but was unable to make a connection. Windows Printers and Scanners would detect the AirSane scanner and request metadata but would not initiate the actual WS-Scan session. I could not find out the reason for this, microsoft and other browsing tools would report correct data transmission. 

As an alternative, the author of this project: https://github.com/roncapat/WSD-python might be prepared to finish his work on the scanner server?

**Comment by stevensantos-docs** [comment]
Any chance I could have a look at the code you wrote?

**Comment by SimulPiscator** [comment]
Sure, but don't expect it to be clear. In its current form, it tries to advertise a WSD printer (but does not succeed in being recognized as such by Windows Printers and Scanners).
[airsaned-wsd.tar.gz](https://github.com/SimulPiscator/AirSane/files/6183907/airsaned-wsd.tar.gz)

**Comment by markosjal** [comment]
Bravo!

**Comment by SimulPiscator** [comment]
Windows 11 now supports eSCL scanners, and works with AirSane. No WSD support needed :-)

**Comment by stevensantos-docs** [comment]
Can you say more about this?  Any links or such?

**Comment by SimulPiscator** [comment]
I found that since the last patchday, you can use Settings->Bluetooth and devices->Printers and scanners to search for AirSane scanners, and install them into the Windows system.

**Comment by stevensantos-docs** [comment]
This is huge.  It means a linux box can now be the central server for
scanners on all platforms

**Comment by markosjal** [comment]
Yes, it is huge, but does AirSane still have no ADF support? 

I know AirSane would not seem to share my Network scanner the last time I tested 
 it,  it only seemed to work with a USB scanner last I tested. 
 
 I Wrote a little PHP code that runs on a Windows 10 (XAMPP Recommended) web host that allows a locally installed Windows scanner to be shared over eSCL , so it even opens back up Windows to Windows scanner sharing (lost some time ago in Windows 10 when homegroup went away) or Windows to Mac or Windows to Linux sharing. The beauty of the Windows solution is that most any scanner that is WIA or TWAIN can be shared regardless of Interface. See http://scannershare.com for more info and download
